### DISCUSSION TITLE
[Feature] Option to automatically resize windows to preset-column-widths

### DISCUSSION THREAD

**Discussion by FlyingWombat** [YaLTeR/niri#385] [Ideas]
I think it would be useful to have an option to force preset-column-widths for all windows, or certain window-rules. By "force" I mean, when a window is created, it will be resized to the closest `preset-column-widths` value. This option should have possible values: "never" (default), "on-creation", and "always". The only issue with "always" would be that it would conflict with `set-column-width` bindings. Specific `window-rules` should override this behavior, if `default-column-width` is set for that rule.

The issue with the current behavior, is that I want to leave `default-column-width {}` because I want some windows big and some windows small, and default sizing generally works for that. But the sizing is sometimes slightly uneven. For example: firefox opens at 60% width, and alacritty opens at 43% width by default. With this option I wouldn't have to make window-rules for every offending application that has uneven default sizing.

**Comment by YaLTeR** [comment]
Hm, I see what you want, it's pretty weird though. If we were to add this, I'd only make it in the form of a flag "let window pick its width but resize to the closest preset width upon open", but I'm not sure I'm entirely convinced to do it.

I personally use window-rule default width overrides for several apps, while it's a bit annoying I don't think it's all that bad.

**Reply by FlyingWombat** [reply]
I agree this is pretty low priority; it would just be a convenience feature for the config. So far I only have a handful of apps that need overrides.

**Comment by morenathan** [comment]
Why not use default-column-width with a default of your choice? Personally I'm autistic so "normal" sizes as I see them are very important. With my default-column-width at 0.5, Firefox obeys quite well.

I follow that up with a preset-column-widths list of 0.25 0.50 0.75 and my life is perfect :) Maybe I missed something but apps don't tent to disobey the proportions.
